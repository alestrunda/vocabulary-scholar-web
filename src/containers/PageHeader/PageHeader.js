import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import classNames from 'classnames';
import IconSyncStatus from '../../components/IconSyncStatus';

export const PageHeader = props => {
  const getTooltipText = () => {
    if (!props.isStateLoaded) return 'Loading state';
    if (!props.isStoreLoadedFromDrive) return 'Google Drive not connected';
    return 'Synchronized with Google Drive';
  };

  return (
    <header
      style={props.style}
      className={classNames('page-header', 'container', props.className)}
    >
      <h1 className="page-header__title text-break">{props.children}</h1>
      <div className="page-header__right">
        <Link data-testid="state-status-link" to="/settings">
          <IconSyncStatus
            className={classNames({ 'is-loaded': props.isStateLoaded })}
            style={{ padding: 0 }}
            iconStyle={{ color: '#fff' }}
            loaderColor="#fff"
            isLoading={!props.isStateLoaded}
            isSynced={props.isStoreLoadedFromDrive}
            tooltip={getTooltipText()}
            tooltipPosition="bottom-left"
          />
        </Link>
      </div>
    </header>
  );
};

PageHeader.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.array,
    PropTypes.string,
  ]).isRequired,
  isStateLoaded: PropTypes.bool.isRequired,
  isStoreLoadedFromDrive: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  isStateLoaded: !!state.app.isStateLoaded,
  isStoreLoadedFromDrive: !!state.gapi.isStoreLoadedFromDrive,
});

export default connect(mapStateToProps)(PageHeader);
