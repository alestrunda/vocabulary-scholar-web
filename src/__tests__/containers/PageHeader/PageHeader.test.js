import React from 'react';
import { shallow } from 'enzyme';

import { PageHeader } from '../../../containers/PageHeader/PageHeader';

it('renders correctly', () => {
  const root = shallow(
    <PageHeader isStateLoaded={false} isStoreLoadedFromDrive={false}>
      Page Title
    </PageHeader>
  );
  expect(root).toMatchSnapshot();
});
