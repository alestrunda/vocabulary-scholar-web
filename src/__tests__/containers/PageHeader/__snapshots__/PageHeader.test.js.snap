// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageHeader
    isStateLoaded={false}
    isStoreLoadedFromDrive={false}
  >
    Page Title
  </PageHeader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="page-header__title text-break"
        >
          Page Title
        </h1>,
        <div
          className="page-header__right"
        >
          <Link
            data-testid="state-status-link"
            to="/settings"
          >
            <IconSyncStatus
              className=""
              iconStyle={
                Object {
                  "color": "#fff",
                }
              }
              isLoading={true}
              isSynced={false}
              loaderColor="#fff"
              style={
                Object {
                  "padding": 0,
                }
              }
              tooltip="Loading state"
              tooltipPosition="bottom-left"
            />
          </Link>
        </div>,
      ],
      "className": "page-header container",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Page Title",
          "className": "page-header__title text-break",
        },
        "ref": null,
        "rendered": "Page Title",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            data-testid="state-status-link"
            to="/settings"
          >
            <IconSyncStatus
              className=""
              iconStyle={
                Object {
                  "color": "#fff",
                }
              }
              isLoading={true}
              isSynced={false}
              loaderColor="#fff"
              style={
                Object {
                  "padding": 0,
                }
              }
              tooltip="Loading state"
              tooltipPosition="bottom-left"
            />
          </Link>,
          "className": "page-header__right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <IconSyncStatus
              className=""
              iconStyle={
                Object {
                  "color": "#fff",
                }
              }
              isLoading={true}
              isSynced={false}
              loaderColor="#fff"
              style={
                Object {
                  "padding": 0,
                }
              }
              tooltip="Loading state"
              tooltipPosition="bottom-left"
            />,
            "data-testid": "state-status-link",
            "to": "/settings",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconStyle": Object {
                "color": "#fff",
              },
              "isLoading": true,
              "isSynced": false,
              "loaderColor": "#fff",
              "style": Object {
                "padding": 0,
              },
              "tooltip": "Loading state",
              "tooltipPosition": "bottom-left",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title text-break"
          >
            Page Title
          </h1>,
          <div
            className="page-header__right"
          >
            <Link
              data-testid="state-status-link"
              to="/settings"
            >
              <IconSyncStatus
                className=""
                iconStyle={
                  Object {
                    "color": "#fff",
                  }
                }
                isLoading={true}
                isSynced={false}
                loaderColor="#fff"
                style={
                  Object {
                    "padding": 0,
                  }
                }
                tooltip="Loading state"
                tooltipPosition="bottom-left"
              />
            </Link>
          </div>,
        ],
        "className": "page-header container",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Page Title",
            "className": "page-header__title text-break",
          },
          "ref": null,
          "rendered": "Page Title",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              data-testid="state-status-link"
              to="/settings"
            >
              <IconSyncStatus
                className=""
                iconStyle={
                  Object {
                    "color": "#fff",
                  }
                }
                isLoading={true}
                isSynced={false}
                loaderColor="#fff"
                style={
                  Object {
                    "padding": 0,
                  }
                }
                tooltip="Loading state"
                tooltipPosition="bottom-left"
              />
            </Link>,
            "className": "page-header__right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <IconSyncStatus
                className=""
                iconStyle={
                  Object {
                    "color": "#fff",
                  }
                }
                isLoading={true}
                isSynced={false}
                loaderColor="#fff"
                style={
                  Object {
                    "padding": 0,
                  }
                }
                tooltip="Loading state"
                tooltipPosition="bottom-left"
              />,
              "data-testid": "state-status-link",
              "to": "/settings",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "iconStyle": Object {
                  "color": "#fff",
                },
                "isLoading": true,
                "isSynced": false,
                "loaderColor": "#fff",
                "style": Object {
                  "padding": 0,
                },
                "tooltip": "Loading state",
                "tooltipPosition": "bottom-left",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
